app/services/party_service.py

```diff
@@ -22,11 +22,14 @@ def get_business_details(ru):
                      text=party_data.text,
                      url=url)
 
-        party_dict = json.loads(party_data.text)
-        if type(party_dict) is list:  # if id is not a uuid returns a list not a dict
-            party_dict = {'errors': party_dict[0]}
-
-        return party_dict, party_data.status_code
+        if party_data.status_code == 200:
+            party_dict = json.loads(party_data.text)
+            if type(party_dict) is list:  # if id is not a uuid returns a list not a dict
+                party_dict = {'errors': party_dict[0]}
+            return party_dict, party_data.status_code
```

Comment:
I don't understand this piece of logic. If the party service responds with a status code of 200 why would we have errors?

Reply:
if type(party_dict) is list:  # if id is not a uuid returns a list not a dict
                 party_dict = {'errors': party_dict[0]}

is now redundant.